apps/prairielearn/src/migrations/20230523200508_enrollment_limits.sql

```diff
@@ -0,0 +1,17 @@
+ALTER TABLE enrollments
+ADD COLUMN IF NOT EXISTS created_at TIMESTAMP WITH TIME ZONE;
+
+-- We set the default as a separate step to avoid setting the value for all
+-- existing rows.
+ALTER TABLE enrollments
+ALTER COLUMN created_at
+SET DEFAULT CURRENT_TIMESTAMP;
+
+ALTER TABLE institutions
+ADD COLUMN IF NOT EXISTS yearly_enrollment_limit INTEGER;
+
+ALTER TABLE institutions
+ADD COLUMN IF NOT EXISTS course_instance_enrollment_limit INTEGER;
+
+ALTER TABLE course_instances
+ADD COLUMN IF NOT EXISTS enrollment_limit INTEGER;
```

Comment:
Can we make all these limits NOT NULL? I'm ok with a high DEFAULT but I want to move towards having limits on everything.

Reply:
For `course_instances.enrollment_limit` specifically, no, I do not want to make this `NOT NULL` - the default should be that this is inherited from the institution (or in the future, the division), which is represented by `NULL`. I _do_ think it makes sense to place `NOT NULL` constraints on the top-level limits on `institutions`. I'll do that. I checked prod data, and I think 10k is a comfortable default limit; that's several times larger than the largest course instance we've ever had.

Tangential but related thought: should limits only be enforced for enterprise servers? The UI to view/edit these limits is only available for enterprise servers, and this seems like an enterprise feature anyways, so this might make sense?

Reply:
Good point about `course_instances.enrollment_limit`. I agree that 10k seems like a fine default and I agree that this limit should only be enforced for EE.